### DISCUSSION TITLE
how to rotate SSL certificates

### DISCUSSION THREAD

**Discussion by littlegreenbean33** [micropython/micropython#15875] [Using MicroPython]
Hi,

I was wondering what the the right way to rotate/renew certificates in mycropython. I have not seen an ACME like library or anything similar.

This is in a controlled, isolated industrial environment with local CA. We use MQTTS for communications but cybersec does not allow to exchange the certificate via MQQT topics.

Any ideas ?

Thanks.

JCG

**Comment by vshymanskyy** [comment]
In general, this is simple if you have an OTA mechanism. Certificates are updated just like any other file. 
Here are [some insights](https://github.com/micropython/micropython/issues/12675#issuecomment-2260239957) on how we've implemented an OTA mechanism for our Blynk products.

**Reply by littlegreenbean33** [reply]
I do not have OTA. Cybersec currently forbids anything like that in the shopfloor. I will be looking to see if there ia an ACME client for micropython

**Reply by vshymanskyy** [reply]
The thing is, updating even a single file, like a certificate on the device via a network, os by definition a OTA update.

If that's not allowed, you should use self signed certificates of other approaches to security.

There's nothing insecure in OTA updates that are implemented properly (and his is not easy).
